File: Assignment 01/Program 08/prog8.py
# Write a C program to merge two unsorted dynamic array in sorted order.

def bubble_sort(a3,n):
    for i in range(n):
        for j in range(n-i-1):
            if(a3[j]>a3[j+1]):
                t = a3[j]
                a3[j] = a3[j+1]
                a3[j+1] = t

import array as arr

a1 = arr.array('i')
n = int(input("Enter the number of inputs: "))
for i in range(n):
    val = int(input(f"Enter the element{i+1}: "))
    a1.append(val)

a2 = arr.array('i')
m = int(input("Enter the number of inputs: "))
for i in range(m):
    val = int(input(f"Enter the element{i+1}: "))
    a2.append(val)

a3 = arr.array('i')
for i in range(n):
    a3.append(a1[i])

for j in range(m):
    a3.append(a2[j])

bubble_sort(a3,(n+m))


print(a3)

File: Assignment 01/Program 08/prog8.c
// Write a C program to merge two unsorted dynamic array in sorted order.

#include<stdio.h>
#include<stdlib.h>

void initialize (int *arr,int n){
    for (int i = 0; i < n; i++)
    {
        printf("Enter the element %d: ",i+1);
        scanf("%d",&arr[i]);
    }
}

void bubble_sort(int *arr, int n){
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n-i-1; j++)
        {
            if(arr[j]>arr[j+1]){
                int t = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = t;
            }
        }
        
    }
    
}
int main(){
    int n ,m;
    printf("Enter the number of inputs for first array: ");
    scanf("%d",&n);
    int *arr1 = (int*)malloc(n*sizeof(int));
    initialize (arr1,n);


    printf("Enter the number of inputs for second array: ");
    scanf("%d",&m);
    int *arr2 = (int*)malloc(m*sizeof(int));
    initialize (arr2,m);
    
    int *new_arr = (int*)malloc((n+m)*sizeof(int));
    for (int k = 0; k < n; k++)
    {
        new_arr[k] = arr1[k];
    }
    for (int k = n,i=0; k < n+m; k++,i++)
    {
        new_arr[k] = arr2[i];
    }

    bubble_sort(new_arr,(n+m));

    for (int i = 0; i < n+m; i++)
    {
        printf("%d ",new_arr[i]);
    }
    
    free(arr1);
    free(arr2);

    return 0;
}